src/libraries/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -568,6 +559,14 @@ public long GetOffset(SequencePosition position)
             }
             else
             {
+                // Verify position validity, this is not covered by BoundsCheck for Multi-Segment Sequence
+                // BoundsCheck for Multi-Segment Sequence check only validity inside current sequence but not for SequencePosition validity.
+                // For single segment position bound check is implicit.
+                Debug.Assert(positionSequenceObject != null);
+
+                if (((ReadOnlySequenceSegment<T>)positionSequenceObject).Memory.Length - positionIndex < 0)
```

Comment:
@ahsonkhan I don't know if this is ok, default BoundCheck method doesn't check for `SequencePosition` validity, but only if offset specified is valid against current ROSequence.
It's implicit for single segment because the object is the same(if sequence position is pertinent).